### DISCUSSION TITLE
Error when saving  Repeater relationship

### DISCUSSION THREAD

**Discussion by alexchugun** [filamentphp/filament#13501] [Help]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

Hello everyone
I made a form and it displays everything perfectly, including related tables:

```php
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('name'),
                Forms\Components\TextInput::make('url')->url(),
                Forms\Components\Toggle::make('is_active'),
                Forms\Components\DatePicker::make('date_end'),
                Forms\Components\Repeater::make('Rules')
                    ->relationship('rules')
                    ->schema([
                        Forms\Components\Select::make('ruleType')
                            ->relationship('ruleType', 'name'),
                        Forms\Components\TextInput::make('interval'),
                        Forms\Components\Toggle::make('is_active')
                    ])
            ]);
    }
```

 but an error appears when saving:

`SQLSTATE[23502]: Not null violation: 7 ERROR: null value in column "rule_type_id" of relation "rules" violates not-null constraint DETAIL: Failing row contains (13, 2024-07-08 14:51:41, 2024-07-08 14:51:41, null, 5, f, 4, null, null).`

What could be the reason?

**Comment by micraux** [comment]
Hi,
If I read correctly the message, the `rule_type_id` contains NULL in your request, but you have set the column to be NOT NULL in your database.
